src/Dscp/Crypto/Encrypt.hs

```diff
@@ -143,19 +156,37 @@ prepareAEAD (PassPhrase pp) =
        aeadInit aeadMode cipher initIV
 
 -- | Encrypt given 'ByteArray' with AES.
-encrypt :: ByteArray ba => PassPhrase -> ba -> Encrypted ba
+encrypt :: FromByteArray ba => PassPhrase -> ba -> Encrypted ba
 encrypt pp plaintext =
     let aead = prepareAEAD pp
-    in uncurry Encrypted $
-       aeadSimpleEncrypt aead authHeader plaintext authTagLength
+        -- TODO: Subject to optimization. For now using converion to bytestring
+        -- since @ba@ is not instance of 'ByteArray'.
+        plaintextBS = BA.convert plaintext
+        (authtag, ciphertextBS :: ByteString) =
+            aeadSimpleEncrypt aead authHeader plaintextBS authTagLength
+        ciphertext =
+            leftToPanicWith "encrypt: got malformed item: " $
+            fromByteArray ciphertextBS
+    in  Encrypted authtag ciphertext
```

Comment:
What is the motivation for these changes? :thinking: If it's only to allow for using `encrypt` directly on `SecretKey` datatype, then it doesn't seem like the most elegant way to perform it (if you don't want to think of fancy ways to avoid conversions, then you might've renamed existing functions to `encryptBA/decryptBA` and make `encrypt/decrypt` function a simple wrappers which perform back and forth conversion.

Reply:
I tried to do conversion in smarter way like adding a wrapper and `FromByteArray ba => ByteArray (PotentiallyMalformed ba)` for it, but it seemed too painful because `ByteArray` requires `Monad` and `ByteArrayAccess`. Would be happy to hear another way to workaround that.